sleap/io/convert.py

```diff
@@ -117,36 +118,42 @@ def main(args: list = None):
             video_search=video_callback,
             video=video_path,
         )
-
-    if args.format == "analysis":
-        from sleap.info.write_tracking_h5 import main as write_analysis
-
-        output_paths = [path for path in args.outputs]
-
-        # Generate filenames if user has not specified (enough) output filenames
-        labels_path = args.input_path
-        fn = re.sub("(\\.json(\\.zip)?|\\.h5|\\.slp)$", "", labels_path)
-        fn = PurePath(fn)
-        default_names = [
-            default_analysis_filename(
-                labels=labels,
-                video=video,
-                output_path=str(fn.parent),
-                output_prefix=str(fn.stem),
-            )
-            for video in labels.videos[len(args.outputs) :]
-        ]
-
-        output_paths.extend(default_names)
-
-        for video, output_path in zip(labels.videos, output_paths):
-            write_analysis(
+    if "analysis" in args.format:
+        vids = []
+        if len(args.video) > 0:  # if a video is specified
+            for v in labels.videos:  # check if it is among the videos in the project
+                if args.video in v.backend.filename:
+                    vids.append(v)
```

Comment:
`args.video` currently is created s.t. it only allows one argument meaning we can break out of this loop early if we wanted to (might be beneficial for datasets that have ample `SingleImageVideo`s)
https://github.com/talmolab/sleap/blob/19b95037cd56ffcc367bd16e6b949c2640aa466f/sleap/io/convert.py#L75-L77

Reply:
@roomrys Thanks for your comments. Not sure, if I understand this one correctly, the suggested implementation should not change the previous behaviour. If the user specifies a video, only that video will be exported, if none is specified, all vids will be exported. Is that not the intended behaviour?

Reply:
Sorry for the confusion. I realize you are not trying to change the functionality of `args.video`. To be more clear, I was suggesting we add a `break` after we find `args.video in v.backend.filename` which would be beneficial in a project with many videos.
```suggestion
            for v in labels.videos:  # check if it is among the videos in the project
                if args.video in v.backend.filename:
                    vids.append(v)
                    break
```

Reply:
Sorry, did this get resolved as well? Seems like a small proposed change.